### ISSUE TITLE
Accessibility Violations (EU law): showDatePicker lacks descriptive TalkBack labels for "Previous" and "Next month" buttons

### ISSUE THREAD

**Issue by BabarShamsi-mset1** [flutter/flutter#171071]
**Steps to Reproduce**

    Enable TalkBack on an Android device.

    In a Flutter app, use the built-in showDatePicker function to open a calendar picker.

    Swipe to the navigation arrows (previous and next month buttons) using TalkBack.

**Expected Behavior**

    In native Android date pickers, TalkBack announces:

    Left arrow: “Switch to previous month button”

    Right arrow: “Switch to next month button”

   These labels provide clear context for screen reader users.

**Actual Behavior**

   In Flutter’s showDatePicker:

    The left and right arrow buttons are only announced as "Button" Or "Button, Disabled" with no descriptive label.

    These internal buttons are not exposed via Semantics and cannot be annotated or customized externally.

    As a result, visually impaired users can't understand their function.

<img width="486" alt="Image" src="https://github.com/user-attachments/assets/39cd9f4c-b571-43c8-b5e9-02277c1b113b" />

<img width="488" alt="Image" src="https://github.com/user-attachments/assets/8e63638c-5b6c-4f5c-93b5-2643b802cf7f" />

**Impact**

    This is critical to be addressed because of EU laws kicking in on 28th June

    This creates a significant accessibility barrier.

    Violates WCAG 4.1.2 (Name, Role, Value).

    Affects user experience for apps that rely on date input.

**Test Environment**

    Flutter version: 3.22.0

    Device: Pixel 8

    OS: Android 15

    Accessibility Service: TalkBack

**Proposed Solution**

    Internally apply Semantics labels to the arrow buttons:

    Left: "Switch to previous month button"

    Right: "Switch to next month button"

**Workaround**

    No current workaround is available. These buttons are not externally exposed or modifiable.

    Optionally expose a parameter to customize or localize these labels.

**Additional Context**

The native Android date picker handles this correctly. Flutter’s version should match this level of accessibility to ensure    inclusiveness and compliance with accessibility standards.

**Comment by tirth-patel-nc** [comment]
# Triage Report

I was able to reproduce the issue using the provided code sample on the `stable (3.29.3)`, `stable (3.32.5)` and `master (3.33.0-1.0.pre.697)` channels. Tested on Android (Pixel 6a Android 16).

Similar issue was reported in https://github.com/flutter/flutter/issues/153562#issuecomment-2532210843. Also #155224 might be related.

https://github.com/user-attachments/assets/4dd6ec22-34cc-417b-9046-98d6a0fd318e

I can observe similar behaviour on macOS as well.

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/a9dcb8e2-b57d-405d-819a-4dba2078f466" />
